WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalNonRetryableError.swift

```diff
@@ -40,13 +43,15 @@ final class CardPresentModalNonRetryableError: CardPresentPaymentsModalViewModel
         return topTitle + bottomTitle
     }
 
-    init(amount: String, error: Error) {
+    init(amount: String, error: Error, onDismission: @escaping () -> Void) {
         self.amount = amount
         self.error = error
+        self.onDismission = onDismission
     }
 
     func didTapPrimaryButton(in viewController: UIViewController?) {
         viewController?.dismiss(animated: true)
+        onDismission()
```

Comment:
nit: maybe this can be called after `viewController?.dismiss` completes in the completion block (default to be `nil`)?

Reply:
That makes more sense indeed, changed in https://github.com/woocommerce/woocommerce-ios/pull/6659/commits/79c975d9df2f7896a4880575d1cabc53e1a19c3b 🚀